tests/integration/test_deduplication.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import json
+
+from helpers import get_prometheus_active_targets, oci_image
+from pytest_operator.plugin import OpsTest
+
+prometheus_app_name = "prometheus"
+prometheus_resources = {"prometheus-image": "prom/prometheus:v2.35.0"}
```

Comment:
Should probably use `upstream-source` here.

Reply:
Done.